cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerModificationService.java

```diff
@@ -274,6 +286,91 @@ public void upgradeClusterRuntime(Set<ClusterComponent> components, boolean patc
         }
     }
 
+    private void tagHostsWithHostTemplateName() throws ApiException, CloudbreakException {
+        ClouderaManagerRepo clouderaManagerRepoDetails = clusterComponentProvider.getClouderaManagerRepoDetails(stack.getCluster().getId());
+        if (!isVersionNewerOrEqualThanLimited(clouderaManagerRepoDetails::getVersion, CLOUDERAMANAGER_VERSION_7_6_0)) {
+            LOGGER.info("Skipping host tagging. Cloudera Manager version needs to be equal or higher, than 7.6.0.");
+            return;
+        }
+        LOGGER.info("Tagging hosts after runtime upgrade.");
+        ApiClient v46Client = buildv46ApiClient();
+        ClustersResourceApi clustersResourceApi = clouderaManagerApiFactory.getClustersResourceApi(v46Client);
+        HostsResourceApi hostsResourceApi = clouderaManagerApiFactory.getHostsResourceApi(v46Client);
+        String clusterName = stack.getName();
+        Set<String> hostnamesFromCM = getClusterHostnamesFromCM(clustersResourceApi, clusterName);
+        Set<Call> callSet = Optional.of(stack).map(Stack::getInstanceGroups).stream().flatMap(Set::stream)
+                .map(asyncTagHostsInHostGroup(hostsResourceApi, hostnamesFromCM))
+                .flatMap(Set::stream)
+                .collect(Collectors.toSet());
```

Comment:
this could be a method:
```
    private Set<Call> startAsyncTagCalls(Set<String> hostnamesFromCM, HostsResourceApi hostsResourceApi) {
        return Optional.of(stack).map(Stack::getInstanceGroups).stream().flatMap(Set::stream)
                .map(asyncTagHostsInHostGroup(hostsResourceApi, hostnamesFromCM))
                .flatMap(Set::stream)
                .collect(Collectors.toSet());
    }
```